onnx/external_data_helper.py

```diff
@@ -78,6 +79,9 @@ def set_external_data(tensor,  # type: TensorProto
                       checksum=None,  # type: Optional[Text]
                       basepath=None  # type: Optional[Text]
                       ):  # type: (...) -> None
+    if not tensor.HasField("raw_data"):
+        raise ValueError("Tensor " + tensor.name + "does not have raw_data field. Cannot set external data for this tensor.")
+
     del tensor.external_data[:]
```

Comment:
Maybe we should clean `tensor.raw_data` as well because it's being set to external

Reply:
> Maybe we should clean `tensor.raw_data` as well because it's being set to external

Reply:
right now converting a model to external data is done in 2 steps. In the first step tensors are simply marked to use external data and in the second step the data is actually serialized. This method is called in both the steps. So we cant simply clean the raw_data here...